etc/helm/pachyderm/templates/pachd/config-job.yaml

```diff
@@ -39,6 +39,48 @@ spec:
         - name: PACH_ADDR
           value: "grpc://pach-enterprise:1650"
         {{- end }}
+        {{- if .Values.pachd.rootTokenSecretName }}    
+        - name: ROOT_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.pachd.rootTokenSecretName }}
+              key: rootToken
+        {{- end }}
+        {{- if .Values.pachd.enterpriseRootTokenSecretName }}
+        - name: ENTERPRISE_ROOT_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.pachd.enterpriseRootTokenSecretName }}
+              key: enterpriseRootToken
+        {{- end }}
+        {{- if .Values.pachd.enterpriseSecretSecretName }}
+        - name: ENTERPRISE_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.pachd.enterpriseSecretSecretName }}
+              key: enterpriseSecret
+        {{- end }}
+        {{- if .Values.pachd.oauthClientSecretSecretName }}
+        - name: PACHD_OAUTH_CLIENT_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.pachd.oauthClientSecretSecretName }}
+              key: pachdOauthClientSecret
+        {{- end }}
+        {{- if .Values.pachd.enterpriseLicenseKeySecretName }}
+        - name: ENTERPRISE_LICENSE_KEY
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.pachd.enterpriseLicenseKeySecretName }}
+              key: enterpriseLicenseKey
+        {{- end }}   
+        {{- if .Values.oidc.upstreamIDPsSecretName }}
+        - name: UPSTREAM_IDPS
```

Comment:
Isn't this an array? What would a configuration here look like?

Reply:
configuration would look the same as specified in the values.yaml for oidc.upstreamIDPs. It's a YAML array. Since it all gets encoded into bytes for the secret, we just deserialize it in the config-pod into YAML when it's read

Reply:
We may want to test this, (I'm not sure new lines and env vars are going to play nicely here?)

Reply:
Tested this locally 👍